### PR TITLE
Preferred data source for a given command and help docs for any command

### PR DESCRIPTION
**Pull request by stkerr**
# Description

- [x] Summary of the change / bug fix.
- [x] Link # issue, if applicable.
- [x] Screenshot of the feature or the bug before/after fix, if applicable.
- [x] Relevant motivation and context. 
- [x] List any dependencies that are required for this change.

This PR adds a new feature settings under `/settings` called `preferred_data_source_file`. It is defaulted to `sources.json`. This PR updates the `/stocks/load` command to use the preferred data source if one is not specified explicitly.

This PR also introduces a new helper function called `get_preferred_source` which can be used to retrieve the preferred value for a given command. If a specific value is not present, the function wlll return the most specific preference as possible, eventually resulting in the default value. In practice, this means a call like `get_preferred_source("stocks.subcommand.subsubcommand")` will return first the value for `stocks.subcommand.subsubcommand`, then `stocks.subcommand["default"]`, then `stocks["default"]`, then eventually the root-level `["default"]` value.

This is intended to allow different parts of Terminal to know what data source the user wants to use and then they can take action based on that.  The individual commands implemented will be responsible for examining and respecting this value.

In the future, this can be used to change the logic inside of various functions depending on which data source the user prefers, rather than hardcoding to Yahoo finance as is currently done in several parts of the Terminal. As various other data sources, such as new brokers, are added, users can select those as their preferred data source instead of Yahoo.

# How has this been tested?

Existing pytests were run successfully.

Several manual runs to change the values in the source file and then run the load command were done.

# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

**Comment by DidierRLopes** [comment]
Hey @stkerr ,

I am currently away until next week, but this is something I wanted for us to tackle in a more generic way.

I was thinking about doing something along the lines of having a file with:
```
{
    'stocks/load': 'yf',
    'crypto/load`: 'cg',
    ....
}
```

And adding a command to access this file and change it in settings. Something where we could do:
`source --cmd stocks/load --value av` to change 1 value
`source` to display the dictionary in a nice way

Thoughts?

File: tests/openbb_terminal/test_settings_controller.py [hunk 1]
```diff
@@ -17,6 +17,11 @@ def controller(mocker):
     return SettingsController()
 
 
+@pytest.mark.parametrize("other", [["sources.json.default"], ["-v", "sources.json.default"]])
+def test_preferred_data_source_file(controller, other):
+    controller.call_preferred_data_source(other)
+
```
Comment (stkerr) [inline tests/openbb_terminal/test_settings_controller.py:23]:
```suggestion
```
Remove double blank line

**Comment by stkerr** [comment]
@DidierRLopes @jmaslek I changed the approach based on the feedback and our discussion in Discord. Can you take a look at this PR? Thanks!

**Comment by DidierRLopes** [comment]
> @DidierRLopes @jmaslek I changed the approach based on the feedback and our discussion in Discord. Can you take a look at this PR? Thanks!

On it!

**Comment by DidierRLopes** [comment]
@stkerr see my PR on your branch: https://github.com/stkerr/OpenBBTerminal/pull/1

Let me know your thoughts, and we can have a chat to discuss my changes 😄

**Review by DidierRLopes — APPROVED** [review]

**Pull request merged**